### PR TITLE
Qradar correlation with polling

### PR DESCRIPTION
**Pull request by orlichter1**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/16382
fixes: https://github.com/demisto/etc/issues/23557

## Description
Change Qradar offensecorrelation playbook to be generic polling. Also add an option to add custom fields.

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No - but deprcate old playbook

## Must have
- [ ] Tests
- [ ] Documentation 

## Files for Tech Review:
new playbooks:
- [ ] [Qradar Correlation Log](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadarCorrelationLog_.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)
- [ ] [Qradar new correlation playbook](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_v2.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)

deprecated:
- [ ] [Qradar old correlations](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)
- [ ] [Deprecated script](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/QRadarGetOffenseCorrelations.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)

new script:
- [ ] [ChangeContext script](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/ChangeContext/ChangeContext.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)

release notes:
- [ ] [Changelog of deprecated correlation playbook](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations__CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)
- [ ] [Changelog of deprecated scripts](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetCorrelationLogs/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)
- [ ] [Another deprecated script](https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341)

### CODE REVIEW

**Comment by yaakovi** [comment]
let's do a short demo for this one

File: Packs/Legacy/Scripts/ChangeContext/CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
+
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/ChangeContext/CHANGELOG.md:2]:
```suggestion
-
```

File: Packs/Legacy/Scripts/QRadarGetCorrelationLogs/QRadarGetCorrelationLogs.yml [hunk 2]
```diff
@@ -8,6 +8,7 @@ tags:
 - QRadar
 comment: Return the QRadar Correlation logs if exist
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/QRadarGetCorrelationLogs/QRadarGetCorrelationLogs.yml:9]: update comment to deprecation standard (see templates in confluence)

File: Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/QRadarGetOffenseCorrelations.yml [hunk 3]
```diff
@@ -8,6 +8,7 @@ tags:
 - QRadar
 comment: Return the QRadar offense correlations if exist in logs.
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/QRadarGetOffenseCorrelations.yml:9]: same

**Comment by ronykoz** [comment]
@orlichter1 what's the status here?

**Comment by orlichter1** [comment]
Doing some changes according to @yaakovi 's PR (which was done on Zoom as we did a demo)will finish in the next few hours.
As you can see in the issue, the first version was sent to the client and approved to be working.

**Comment by orlichter1** [comment]
@ronykoz Also fixing @idovandijk's issue

**Comment by ronykoz** [comment]
👍 Ping me once it's ready I will review it

**Comment by orlichter1** [comment]
@ronykoz Ping.
We can do a short zoom call if you want as to what's going on as this is a playbook.
In the meantime, I'm working no the README files

File: Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_.yml [hunk 4]
```diff
@@ -2,16 +2,9 @@ id: 'QRadar - Get offense correlations '
...
-description: |-
-  Run on a QRadar offense to get more information
-
-  * Get all correlations relevant to the offense
-  * Get all logs relevant to the correlations (not done by default, set "GetCorrelationLogs" to "True")
-
-  Inputs:
-  * GetCorrelationLogs (default: False)
-  * MaxLogsCount (default: 20)
+description: "Deprecated. Use `QRadar - Get offense correlations v2` instead."
```
Comment (ronykoz) [inline Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_.yml:5]: keep the old info here too

File: Packs/Legacy/Scripts/ChangeContext/ChangeContext.py [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+                return key.title()
+            return key
+
+    new_context = createContext(context, keyTransform=replace_func)
+
+    if inplace:
+        demisto.executeCommand("Set", {'key': output_key, 'value': new_context})
+        return f"Changed {output_key} successfully", {}, {}
+    else:
+        return f"Appended {output_key} successfully", {output_key: new_context}, {}
```
Comment (ronykoz) [inline Packs/Legacy/Scripts/ChangeContext/ChangeContext.py:33]: might be missing something but where did you append

File: Packs/Legacy/Scripts/ChangeContext/ChangeContext_test.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+def test_capitalize_context_list_not_inplace(mocker):
+    context = [{'key': 'val', 'key2': 'val2'},
+               {'list2key1': 1, 'list2key2': 2}]
+    args = {'input': context, 'output_key': 'Demisto.Test', 'capitalize': 'True', 'inplace': 'False'}
+    _, ec, _ = replace_context(args)
+
+    assert ec == {'Demisto.Test': [{'Key': 'val', 'Key2': 'val2'}, {'List2Key1': 1, 'List2Key2': 2}]}
+
+
+def test_change_context_path_inplace(mocker):
```
Comment (ronykoz) [inline Packs/Legacy/Scripts/ChangeContext/ChangeContext_test.py:27]: add a test with inplace false and replace dict

File: Packs/Legacy/Scripts/ChangeContext/ChangeContext.py (hunk 5 above)
Comment (orlichter1) [reply Packs/Legacy/Scripts/ChangeContext/ChangeContext.py:33]: I output it to context without "Set"-ing.

**Comment by orlichter1** [comment]
@kirbles19 could you please go over this :)

File: Packs/Legacy/Playbooks/playbook-QRadarCorrelationLog_CHANGELOG.md [hunk 7]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
```
Comment (yaakovi) [inline Packs/Legacy/Playbooks/playbook-QRadarCorrelationLog_CHANGELOG.md:1]: remove this file

File: Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_v2_CHANGELOG.md [hunk 8]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
```
Comment (yaakovi) [inline Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_v2_CHANGELOG.md:1]: remove this file

File: Packs/Legacy/Scripts/ChangeContext/CHANGELOG.md [hunk 9]
```diff
@@ -0,0 +1,2 @@
+## [Unreleased]
```
Comment (yaakovi) [inline Packs/Legacy/Scripts/ChangeContext/CHANGELOG.md:1]: remove this file

File: Packs/Legacy/Scripts/QRadarGetCorrelationLogs/QRadarGetCorrelationLogs.yml (hunk 2 above)
Comment (yaakovi) [reply Packs/Legacy/Scripts/QRadarGetCorrelationLogs/QRadarGetCorrelationLogs.yml:10]: ?

File: TestPlaybooks/playbook-QRadarCorreltaions_v2.yml [hunk 10]
```diff
@@ -0,0 +1,149 @@
...
+    task:
+      id: 37a8922f-0558-4dfd-8f28-4c0a65f43c95
+      version: -1
+      name: Validate context
+      type: condition
+      iscommand: false
+      brand: ''
+      description: ''
+    nexttasks:
+      yes:
```
Comment (yaakovi) [inline TestPlaybooks/playbook-QRadarCorreltaions_v2.yml:86]: add an else flow with `PrintErrorEntry`

**Comment by yaakovi** [comment]
@yaron-libman 
please review those files:
* https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadarCorrelationLog_.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341
* https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar_-_Get_offense_correlations_v2.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

@kirbles19 
please review script yml:
https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/ChangeContext/ChangeContext.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

**Comment by kirbles19** [comment]
@orlichter1 Does anything else have to be reviewed besides the changelogs?

**Comment by ronykoz** [comment]
@kirbles19 i think these are the relevant files:
https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar-GetOffenseCorrelationsV2.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadar-GetOffenseCorrelations_CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Playbooks/playbook-QRadarCorrelationLog.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/ChangeContext/ChangeContext.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetCorrelationLogs/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

https://github.com/demisto/content/edit/Qradar_correlation_2_playbooks/Packs/Legacy/Scripts/QRadarGetOffenseCorrelations/QRadarGetOffenseCorrelations.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F6341

**Comment by kirbles19** [comment]
@orlichter1 Done with everything except the playbooks. 
@yaron-libman When done with playbooks, please add the `docs-approved` label

**Comment by yaron-libman** [comment]
Done

**Comment by ronykoz** [comment]
@orlichter1 let's fix the conflicts and merge this

**Pull request merged**